### PR TITLE
Add unpack_int4 kernel

### PR DESCRIPTION
**Pull request by music-dino**
- Implement JIT kernel for unpack_int4
- Remove unpack_int4 CPU offloading for GPU target

### CODE REVIEW

File: src/targets/gpu/kernels/include/migraphx/kernels/unpack_int4.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const auto input_shape = input.get_shape();
+
+    make_index().global_stride(input_shape.elements(), [&](auto i) {
+        auto idx = input_shape.multi(i);
+        idx[Axis] *= 2;
+        const auto input_val = input[i];
+
+        // unpack_int4 op's normalize_compute_shape will ensure that Input::type is either uint8_t
+        // or int8_t
+        if constexpr(is_same<typename Input::type, uint8_t>::value)
```
Comment (pfultz2) [inline src/targets/gpu/kernels/include/migraphx/kernels/unpack_int4.hpp:45]: Dont use `::value`, use the constructor instead. It also would probably be better to use `is_unsigned<typename Input::type>{}` as well.

Comment (music-dino) [reply src/targets/gpu/kernels/include/migraphx/kernels/unpack_int4.hpp]: Done

File: test/ref/unpack_int4.cpp [hunk 2]
```diff
@@ -125,3 +125,23 @@ TEST_CASE(unpack_int4_nchw)
...
 }
+
+TEST_CASE(unpack_int4_signed)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape s{migraphx::shape::int8_type, {2, 2}};
+    auto l0 = mm->add_literal(migraphx::literal{
+        s,
+        {0b1000'0111 /*-8'7*/, 0b0001'1111 /*-1'1*/, 0b1110'1001 /*-2'-7*/, 0b0101'0111 /*5'7*/}});
```
Comment (lakhinderwalia) [inline test/ref/unpack_int4.cpp:136]: `0b0001'1111 /*-1'1*/` : 1, -1.

File: test/verify/main.cpp [hunk 3]
```diff
@@ -116,6 +116,10 @@ int main(int argc, const char* argv[])
             // disabled because CPU does eliminate_data_type to float for everything
             "test_bitwise_and<migraphx::shape::int32_type>",
             "test_bitwise_and<migraphx::shape::uint8_type>",
+            "test_unpack_int4<migraphx::shape::uint8_type>",
```
Comment (lakhinderwalia) [inline test/verify/main.cpp:119]: Please add a blank line (or a comment) above this line. This is a different test, after all. Thanks.

**Review by lakhinderwalia — APPROVED** [review]
Left a couple of minor comments. Approved.

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**